### PR TITLE
[service.upnext] v1.0.0

### PR DESCRIPTION
**Pull request by im85288**
### Description
Up Next is a replacement for service.nextup.notification
### Checklist:

- [ x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [ x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [ x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by Razzeee** [comment]
Little headsup, I triggered a rerun of our CI, to get you a correct report :)

**Comment by im85288** [comment]
Thanks :) It’s all good as far as I can see..with the exception of that one bug in the build script?

File: service.upnext/addon.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version='1.0' encoding='UTF-8' standalone='yes'?>
+<addon id="service.upnext" name="Up Next" provider-name="im85288" version="1.0.0">
+    <requires>
+        <import addon="xbmc.python" version="2.26.0"/>
+        <import addon="script.module.requests" version="2.9.1" />
+        <import addon="script.module.simplejson" version="3.3.0"/>
+    </requires>
+    <extension library="service.py" point="xbmc.service" start="login"/>
```
Comment (Razzeee) [inline service.upnext/addon.xml]: Please remove `start="login"` as this was removed for leia. It will always start on login by default now.

**Comment by Razzeee** [comment]
I commented on the relevant part from the automatic review.

**Comment by im85288** [comment]
Ahh I see it now! Thanks I’ll get that fixed and resubmitted this weekend.

File: service.upnext/addon.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    <extension point="xbmc.addon.metadata">
+        <summary lang="en">Provides a netflix style pop up notification for watching the next episode with fancy/simple modes.</summary>
```
Comment (enen92) [inline service.upnext/addon.xml]: en_GB

File: service.upnext/addon.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        <description lang="en">Provides a netflix style pop up notification for watching the next episode with fancy/simple modes. A prompt to check if your still watching also occurs after a few automatically watched shows in a row.</description>
```
Comment (enen92) [inline service.upnext/addon.xml]: en_GB

File: service.upnext/addon.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        <platform>all</platform>
+        <language/>
+    <assets>
```
Comment (enen92) [inline service.upnext/addon.xml]: since you're still to make changes to the code: this could be formated

File: service.upnext/addon.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        <import addon="xbmc.python" version="2.26.0"/>
+        <import addon="script.module.requests" version="2.9.1" />
+        <import addon="script.module.simplejson" version="3.3.0"/>
+    </requires>
+    <extension library="service.py" point="xbmc.service" start="login"/>
+    <extension point="xbmc.addon.metadata">
+        <summary lang="en">Provides a netflix style pop up notification for watching the next episode with fancy/simple modes.</summary>
+        <description lang="en">Provides a netflix style pop up notification for watching the next episode with fancy/simple modes. A prompt to check if your still watching also occurs after a few automatically watched shows in a row.</description>
+        <platform>all</platform>
+        <language/>
```
Comment (enen92) [inline service.upnext/addon.xml]: please drop language tag, not applied to script addons that do not provide content

File: service.upnext/resources/lib/monitor.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import xbmc
+import resources.lib.utils as utils
+from resources.lib.player import Player
+
+class Monitor(xbmc.Monitor):
+
+    def __init__(self, *args):
+        self.logMsg("Starting UpNext Service", 0)
+        self.logMsg("========  START %s  ========" % utils.addon_name(), 0)
```
Comment (enen92) [inline service.upnext/resources/lib/monitor.py]: most of those log lines are just spam. The kodi log will already include most of the info here

File: service.upnext/resources/lib/player.py [hunk 7]
```diff
@@ -0,0 +1,424 @@
...
+    import simplejson as json
+else:
+    import json
+
+# service class for playback monitoring
+class Player(xbmc.Player):
+    # Borg - multiple instances, shared state
+    _shared_state = {}
+
+    xbmcplayer = xbmc.Player()
```
Comment (enen92) [inline service.upnext/resources/lib/player.py]: do you need this? Player itself is an instance of `xbmc.Player`

File: service.upnext/resources/lib/player.py [hunk 8]
```diff
@@ -0,0 +1,424 @@
...
+        self.__dict__ = self._shared_state
+        self.logMsg("Starting playback monitor service", 1)
+        self.setupFromSettings()
+        xbmc.Player.__init__(self)
+
+    def logMsg(self, msg, lvl=1):
+        self.className = self.__class__.__name__
+        utils.logMsg("%s %s" % (utils.addon_name(), self.className), msg, int(lvl))
+
+    def onPlayBackStarted(self):
```
Comment (enen92) [inline service.upnext/resources/lib/player.py]: you probably mean onAVStarted

File: service.upnext/resources/lib/player.py [hunk 9]
```diff
@@ -0,0 +1,424 @@
...
+        # Will be called when kodi starts playing a file
+        self.addon_data = {}
+        if utils.settings("developerMode") == "true":
+            self.developerPlayPlayback()
+
+    def getNowPlaying(self):
```
Comment (enen92) [inline service.upnext/resources/lib/player.py]: It looks odd to use the json api to get the video InfoTag in a Player class instance

File: service.upnext/resources/lib/utils.py [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+def decode_data(data):
+
+    data = json.loads(data)
+
+    if data:
+        return json.loads(binascii.unhexlify(data[0]))
+
+
+def logMsg(title, msg, level=1):
+    logLevel = int(settings("logLevel"))
```
Comment (enen92) [inline service.upnext/resources/lib/utils.py]: default for logging should be debug

File: service.upnext/service.py [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import xbmcaddon
+import xbmc
+import os
+from resources.lib.monitor import Monitor
+
+cwd = xbmcaddon.Addon(id='service.upnext').getAddonInfo('path').decode('utf-8')
+BASE_RESOURCE_PATH = xbmc.translatePath(os.path.join(cwd, 'resources', 'lib')).decode('utf-8')
```
Comment (enen92) [inline service.upnext/service.py]: why do you need to append a path that belongs to your addon using the sys module?

File: service.upnext/addon.xml [hunk 12]
```diff
@@ -0,0 +1,23 @@
+<?xml version='1.0' encoding='UTF-8' standalone='yes'?>
+<addon id="service.upnext" name="Up Next" provider-name="im85288" version="1.0.0">
+    <requires>
+        <import addon="xbmc.python" version="2.26.0"/>
+        <import addon="script.module.requests" version="2.9.1" />
+        <import addon="script.module.simplejson" version="3.3.0"/>
```
Comment (Rechi) [inline service.upnext/addon.xml]: Not needed as Leia is Python 2.7 on all platforms, which has the json module.

File: service.upnext/resources/lib/utils.py [hunk 13]
```diff
@@ -0,0 +1,167 @@
+import xbmc
+import xbmcgui
+import xbmcaddon
+import inspect
+import binascii
+import sys
+if sys.version_info < (2, 7):
```
Comment (Rechi) [inline service.upnext/resources/lib/utils.py]: same

File: service.upnext/resources/settings.xml [hunk 14]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
+<settings>
+    <category label="30003">
+        <setting id="simpleMode" type="enum" label="30025" values="Simple|Fancy" default="1"/>
```
Comment (Rechi) [inline service.upnext/resources/settings.xml]: could be localized

File: service.upnext/resources/settings.xml [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+        <setting label="30001" id="autoPlaySeasonTime" type="slider" default="30" range="10,5,60" option="int" />
+        <setting id="autoPlayMode" type="enum" label="30005" values="Play|Don't Play" default="0"/>
```
Comment (Rechi) [inline service.upnext/resources/settings.xml]: same

File: service.upnext/resources/settings.xml [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+    <category label="30004">
+        <setting id="disableNextUp" type="bool" label="30014" default="false" visible="true" enable="true"/>
+        <setting id="logLevel" type="enum" label="30002" values="None|Info|Debug" default="0"/>
+        <setting id="shortPlayMode" type="bool" label="30017" default="false" visible="true" enable="true"/>
+        <setting id="shortPlayLength" type="number" label="30018" default="10" visible="true" enable="true"/>
+        <setting id="shortPlayNotification" type="bool" label="30019" default="true" visible="true" enable="true"/>
+    </category>
+    <category label="30028">
+        <setting id="developerMode" type="bool" label="30029" default="false" visible="true" enable="true"/>
+        <setting id="windowMode" type="enum" label="30030" values="UpNext|UpNextSimple|StillWatching|StillWatchingSimple" visible="eq(-1,true)" default="0"/>
```
Comment (Rechi) [inline service.upnext/resources/settings.xml]: same

**Comment by Rechi** [comment]
> Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

Please squash your commits into a single one with the correct commit message.

**Comment by Rechi** [comment]
To fix your pull request execute the following commands (origin is your repo, upstream is the xbmc repo)

```
git fetch upstream
git checkout im85288
git reset --hard upstream/im85288
git checkout origin/im85288 -- service.upnext/
git commit -m '[service.upnext] v1.0.0'
git push -f origin leia
```

**Comment by im85288** [comment]
I tried those commands with no success, the upstream etc was not working on my command line. Closed in favour of [this new one](https://github.com/xbmc/repo-scripts/pull/947)

**Comment by Rechi** [comment]
`git add remote upstream https://github.com/xbmc/repo-scripts.git`

**Pull request closed without merging**